index.js

```diff
@@ -416,64 +421,122 @@ var msg = [];
       });
     },
 
-
-  ], function(err) {
+  ], function (err) {
     if (err) {
-      debug('Initialization error! '.red, err);
-      // Matrix.device.drivers.led.error();
-      //TODO Connectivity error needs to be handled gracefully
-      // Sample error message in err = 'matrix A network error (such as timeout, interrupted connection or unreachable host) has occurred.'
-      Matrix.haltTheMatrix();
-      if (err.status_code === 400) {
-        return console.error('Incorrect or missing registration information. This device is not correctly configured. Please add MATRIX_DEVICE_ID and MATRIX_DEVICE_SECRET variables. If you do not have these available, you can get them by issuing `matrix register device` with matrix CLI. \n\nIf you continue to have problems, please reach out to our support forums at http://community.matrix.one'.yellow);
+      if (checks.connectivity === false) {
+        console.error('Network error, please make sure you have a valid connection and try again'.yellow);
+        //TODO Try again after a specific time has passed
+        //deviceSetupTimeout = setTimeout(deviceSetup, 60000);
+        // Matrix.device.drivers.led.error();
+        //TODO Connectivity error needs to be handled gracefully
+        // Sample error message in err = 'matrix A network error (such as timeout, interrupted connection or unreachable host) has occurred.'
       } else {
         return error('Bad Matrix Initialization', err);
+        Matrix.haltTheMatrix();
       }
-    }
+    } else {
+      Matrix.service.firebase.device.goOnline();
```

Comment:
load will also error out if firebase / device ID match fails. this will error out here